rfcs/text/design-tokens.md

```diff
@@ -0,0 +1,154 @@
+- Start Date: (fill me in with today's date, 2021-06-14)
+
+
+- [Summary](#summary)
+- [Basic example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed design](#detailed-design)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+    - [Design tokens via styled-components and CSS variables](#design-tokens-via-styled-components-and-css-variables)
+    - [Tag function for generating token](#tag-function-for-generating-token)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+
+# Summary
+
+Implementation of the design tokens with theme support and backwards compatibility.
+
+# Basic example
+
+Design tokens will be kept in `@sage/design-tokens` library. It will be updated every time tokens will change. They will
```

Comment:
Good point - can `@sage/design-tokens` be upgraded by downstream applications themselves or should carbon lock to an explicit version of `@sage/design-tokens` to prevent that (so we can enforce compatibility with carbon)

Reply:
Every update of `@sage/design-tokens` (https://www.npmjs.com/package/@sage/design-tokens) should be compatible backwards. Then updates would be automated so that carbon would contain the latest version of the design tokens. Of course after updating tokens new minor or patch version of carbon would be released.

However I see it this way, that updating design tokens in carbon would be responsibility of the Design Sytem Frontend Team, and they would do it manually to have better control on whats changing. At least at the beginning stage, untill we gather informations on how it would be fully automated.

Reply:
>  should be compatible backwards

Please can you clarify if you mean `will be` or might there be a breaking change at some point?

With this approach it means that some consumers might have to take a breaking change in carbon to get a non breaking change in the design system. We've avoided patching old versions of carbon due to the maintenance overhead but it might be beneficial for DS alignment.

What's your thought's on this?

- Consumer is on v60
- Carbon is on v80
- DS updates pallet and releases v80.0.1

How does Consumer on v60 update, do they have to update to v80 or will you patch v60.0.1?

Reply:
The other question here is - should the design tokens be in a separate repo - or are they just part of carbon?

Reply:
>Please can you clarify if you mean will be compatible backwards or might there be a breaking change at some point?

I mean, that new themes will be usable with all components inside the same version of carbon, since migrating to design tokens will take some time.

> What's your thought's on this?
>
> Consumer is on v60
> Carbon is on v80
> DS updates pallet and releases v80.0.1

Right now, any change of the color, or other values requires releasing new version of the carbon. And this workflow does not have to change.

However with design tokens it will be possible to separate this concern by providing themes to carbon from external source, whether we'll go with CSS o JS approach. Carbon would have only its default theme included.

And how are you dealing with such changes right now?

> The other question here is - should the design tokens be in a separate repo - or are they just part of carbon?

Design Tokens should be kept in separate reposiotry, since they are consumed not only by carbon, but also by libraries on iOS or Android. By definiton they should be consistent across all platforms, and keeping it in a separate repository is a good practice to achieve this.

Reply:
> And how are you dealing with such changes right now?

We follow the same approach today, consumers must be on latest carbon version to get latest theme.

I'm happy for it to stay that way, but it's important that this decision is documented. A simple change to a color, will require taking any breaking changes that have occurred in carbon since the last update.